pyproject.toml

```diff
@@ -0,0 +1,112 @@
+[build-system]
+requires = [
+    "setuptools >= 66",
+    "wheel",
+    "setuptools_scm >= 6.4",
+]
+build-backend = "setuptools.build_meta"
+
+[project]
+authors = [
+  {name = "Jon Haitz Legarreta"}, {email = "jon.haitz.legarreta@gmail.com"}
+]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Science/Research",
+    "Programming Language :: Python :: 3",
+    "Topic :: Scientific/Engineering :: Medical Science Apps.",
+]
+dependencies = [
+    # common dependencies
+    "comet-ml == 3.0.0",
+    "h5py == 2.10.0",
+    "matplotlib >= 2.2.0",
+    "joblib == 1.1.0",
+    "numpy >= 1.20.0",
+    "pandas",
+    "pathos",
+    "pydot",
+    "pygame",
+    "pyyaml",
+    "seaborn == 0.11.0",
+    "scikit-learn == 0.22.0",
+    "scipy == 1.4.0",
+    "torch == 1.8.1",
+    "torchsummary == 1.5.1",
+    "tqdm",
+    "umap-learn",
+    # DWI dependencies
+    "dipy == 1.3.0",
+    "fury == 0.7.1",
+    "nibabel == 3.0.0",
+    # Git dependencies
+    "scilpy @ git+https://github.com/scilus/scilpy@1.2.2",
+]
+description = "Tractography learning"
+dynamic = ["version"]
+keywords = ["DL, dMRI, neuroimaging, tractography"]
+maintainers = [
+  {name = "Jon Haitz Legarreta"}, {email = "jon.haitz.legarreta@gmail.com"}
+]
+name = "tractolearn"
+readme = "README.md"
+requires-python = ">=3.8"
+
+[project.optional-dependencies]
+test = [
+    "hypothesis >= 6.8.0",
+    "pytest == 5.3",
+    "pytest-cov",
+    "pytest-pep8",
+    "pytest-xdist",
+]
+dev = [
+    "black == 22.12",
+    "flake8 == 3.9.2",
+    "flake8-docstrings == 1.6.0",
+    "isort == 5.8.0",
+    "pre-commit >= 2.9.0",
+]
+
+[options.extras_require]
+all = [
+    "%(test)s",
+]
+
+[project.urls]
+homepage = "https://github.com/scil-vital/tractolearn"
+documentation = "https://tractolearn.readthedocs.io/en/latest/"
+repository = "https://github.com/scil-vital/tractolearn"
+
+[tool.black]
+line-length = 79
+target-version = ["py38"]
+exclude ='''
+(
+  /(
+      \.eggs        # exclude a few common directories in the
+    | \.git         # root of the project
+    | \.hg
+    | \.mypy_cache
+    | \.tox
+    | \.venv
+    | _build
+    | buck-out
+    | build
+    | dist
+  )/
+  | data            # also separately exclude project-specific files
+                    # and folders in the root of the project
+)
+'''
+
+[tool.isort]
+profile = "black"
+line_length = 79
+src_paths = ["tractolearn"]
+
+[tool.setuptools.packages]
+find = {}  # Scanning implicit namespaces is active by default
+
+[tool.setuptools_scm]
+write_to = "tractolearn/_version.py"
```

Comment:
Have not figured that out/had left that for future work. IMO could be done separately, but if you know how we can add and test that I'd be glad to include it in this PR.

Reply:
Yes, usually it works fine with by including them in the `setup.py`. With the `pyproject.toml` we can do :[project.scripts]
ae_bundle_streamlines = "scripts:ae_bundle_streamlines.main" 

and it works fine. 
On the other hand, I have been trying to fix a dependency bug with `numpy==1.20` but the package is mandatory for `scilpy1.2.2`

Reply:
OK as for the scripts. Will include that.

As for SCILPY/NumPy, it is always hard to match the SCILPY versions, as they are strict. I think I was forced to remove the version because otherwise i had conflicts with other packages. Even if pip notifies about incompatibilities across versions, we can usually get away with them. Otherwise, we can add a note to the `README` to require a late `pip install -m upgrade numpy==1.20` but I really doubt such version is strictly necessary even for SCILPY.

Reply:
Done for the scripts in ae74d67.

Reply:
Ok maybe we can add a note but on my side I was forced to upgrade to `numpy==1.23.*`. Is there a way to add in the `pyproject.toml`. Else I will accept the PR

Reply:
Not sure if I tried pinning that version. Why should we prefer that version over the latest (currently 1.24.1)? The checks are happy with the current dependencies.

Reply:
The most recent push-force [ee16f8d](https://github.com/scil-vital/tractolearn/commit/ee16f8d1114651daa83171f209ac12297f15d8f7):
- Adds the NumPy version-related note to the `README` file.
- Upgrades accordingly the NumPy version in the GHA workflow file.

Reply:
Resolving.